centos7/Dockerfile.upgrade.centos7

```diff
@@ -1,37 +1,45 @@
 ARG BASEOS
 ARG BASEVER
 ARG PG_FULL
+ARG PG_MAJOR
 ARG PREFIX
 FROM ${PREFIX}/crunchy-pg-base:${BASEOS}-${PG_FULL}-${BASEVER}
 
 LABEL name="upgrade" \
 	summary="Provides a pg_upgrade capability that performs a major PostgreSQL upgrade." \
-	description="Provides a means to perform a major PostgreSQL upgrade from 9.5 to 9.6 or 9.6 to 10. Old data files are left intact." \
+	description="Provides a means to perform a major PostgreSQL upgrade from an earlier version to PostgreSQL ${PG_MAJOR}." \
 	io.k8s.description="postgres upgrade container" \
 	io.k8s.display-name="Crunchy PostgreSQL upgrade container" \
 	io.openshift.tags="postgresql,postgres,upgrade,database,crunchy"
 
-# Add previous version repos as pg-base only covers PG_MAJOR
-ADD conf/crunchypg95.repo /etc/yum.repos.d/
-ADD conf/crunchypg96.repo /etc/yum.repos.d/
+# Add in the repository files with the correct PostgreSQL versions
+ADD conf/crunchypg*.repo /etc/yum.repos.d/
 
+# install the highest version of PostgreSQL + pgAudit and its dependencies as
+# well as unzip
 RUN yum -y install \
 	--setopt=skip_missing_names_on_install=False \
-	pgaudit95 \
-	postgresql95 \
-	postgresql95-contrib \
-	postgresql95-server \
-	pgaudit96 \
-	postgresql96 \
-	postgresql96-contrib \
-	postgresql96-server \
-	pgaudit10 \
-	postgresql10 \
-	postgresql10-contrib \
-	postgresql10-server \
+	--disablerepo=crunchypg* \
+	--enablerepo="crunchypg${PG_MAJOR}" \
+	"postgresql${PG_MAJOR}" \
+	"postgresql${PG_MAJOR}-contrib" \
+	"postgresql${PG_MAJOR}-server" \
+	"pgaudit${PG_MAJOR}*" \
 	unzip \
 	&& yum -y clean all
 
+# add in all of the earlier version of PostgreSQL. It will install the version
+# above, but the dependencies are handled
+RUN yum -y install \
+	--setopt=skip_missing_names_on_install=False \
+	--disablerepo=* \
+	--enablerepo=crunchypg* \
+	postgresql[1-9][0-9] \
+	postgresql[1-9][1-9]-contrib \
+	postgresql[1-9][1-9]-server \
+	pgaudit[1-9][1-9] \
```

Comment:
Should all these number ranges match? Why does the first one have a zero but the others don't?

Reply:
Oh I messed up, the second ones should be `[0-9]` :|